src/app/(nextjs_migration)/(authenticated)/admin/send-test-email/route.ts

```diff
@@ -0,0 +1,115 @@
+import { signupReportEmailPartial } from "../../../../../views/emails/emailSignupReport.js";
+
+export async function POST(req: NextRequest) {
+  const { emailId, recipient } = await req.json();
+  const l10n = getL10n();
+
+  switch (emailId) {
+    case EmailTemplateType.Verification: {
+      // Send test verification email
+      const emailTemplate = getTemplate(
+        getVerificationDummyData(recipient, l10n),
+        verifyPartial,
+        l10n
+      );
+      await initEmail(process.env.SMTP_URL);
+      await sendEmail(
+        recipient,
+        l10n.getString("email-subject-verify"),
+        emailTemplate
+      );
+      break;
+    }
+    case EmailTemplateType.Notification: {
+      // Send test breach notification email
+      await sendTestNotification(req, new NextResponse());
+      break;
+    }
+    case EmailTemplateType.Monthly: {
+      // Send test monthly unresolved breaches email
+      const emailTemplate = getTemplate(
+        getMonthlyDummyData(AppConstants.EMAIL_TEST_RECIPIENT, l10n),
+        monthlyUnresolvedEmailPartial,
+        l10n
+      );
+      await initEmail(process.env.SMTP_URL);
+      await sendEmail(
+        recipient,
+        l10n.getString("email-unresolved-heading"),
+        emailTemplate
+      );
+      break;
+    }
+    case EmailTemplateType.SignupReport: {
+      // Send test sign-up report email
+      const emailTemplate = getTemplate(
+        getSignupReportDummyData(AppConstants.EMAIL_TEST_RECIPIENT, l10n),
+        signupReportEmailPartial,
+        l10n
+      );
+      await initEmail(process.env.SMTP_URL);
+      await sendEmail(
+        recipient,
+        l10n.getString("email-subject-found-breaches"),
+        emailTemplate
+      );
+      break;
+    }
+    default: {
+      throw new Error(`No test email found for ${emailId}`);
+    }
+  }
+
+  console.info(`Sent test email: ${emailId}`);
+
+  // The notify function has its own response
+  if (emailId !== EmailTemplateType.Notification) {
+    return NextResponse.json(
+      { success: true, message: `Sent test ${emailId} email` },
+      { status: 200 }
+    );
+  }
+}
+
+async function sendTestNotification(req: NextRequest, res: NextResponse) {
+  // The test breach notification can be viewed in the public Mailinator inbox
+  // as documented in the README:
+  // https://github.com/mozilla/blurts-server#trigger-breach-alert-email
+  const breachNotificationData = {
+    breachName: "Adobe",
+    // Hash for dummy email `localmonitor20200827@mailinator.com`
+    hashPrefix: "365050",
+    hashSuffixes: ["53cbb89874fc738c0512daf12bc4d91765"],
+  };
+
+  const notifyReq = {
+    app: req.app,
+    body: {
+      ...req.body,
+      ...breachNotificationData,
+    },
+    token: AppConstants.HIBP_NOTIFY_TOKEN,
+  };
+
+  // Disabled for now; not sure yet how to trigger this with the functionality
+  // moved to a Cloud Function.
+  // await notify(notifyReq, res);
```

Comment:
Oh I see, this code wants to call https://github.com/mozilla/blurts-server/blob/1037c75c9361c62ee3c261b66996be308addf8cc/src/controllers/hibp.js#L42, I think we should try to make it work against the utils function. If it's important to keep using the `notify` request handler then we can make that work.